### ISSUE TITLE
Share DataTemplate s across projects

### ISSUE THREAD

**Issue by yll690** [AvaloniaUI/Avalonia#8428]
### Discussed in https://github.com/AvaloniaUI/Avalonia/discussions/8426

<div type='discussions-op-text'>

<sup>Originally posted by **yll690** July  1, 2022</sup>
In WPF I can put the `DataTemplate`s in ResourceDictionary and merge it in other application. But in Avalonia there are many limits:

- `DataTemplate`s can't be put in ResourceDictionary without key. It's also not work when has a key like `vm:TestItem`.
- `DataTemplates` property is readonly.
- `DataTemplates` can not merge each other like `ResourceDictionary` or `Styles`.

So is there any way to share `DataTemplate`s across  projects?</div>

The simplest way seems implements a `DataTemplateInclude`.

**Comment by maxkatz6** [comment]
Answered here https://github.com/AvaloniaUI/Avalonia/discussions/8426#discussioncomment-3067708

Also seems like a duplicate of https://github.com/AvaloniaUI/Avalonia/issues/2688. Except we don't have DataTemplateInclude, if it's even needed.

**Comment by timunie** [comment]
Is there any huge drawback of having this?

Resources > ResourceInclude
Styles > StyleInclude
DataTemplates > ?

**Comment by yll690** [comment]
There's one thing I forget to point out: I can put `DataTemplate` in `ResourceDictionary` and reference it from other projects but it can't apply to models automatically, It must be referenced explicitly by `DynamicResources`. So this `ResourceDictionary`  isn't much helpful in this situation. A `DataTemplateInclude` will be helpful though.

**Comment by trommlbomml** [comment]
> Is there any huge drawback of having this?
> 
> Resources > ResourceInclude Styles > StyleInclude DataTemplates > ?

Yes, because you cannot use the very handy feature defining data templates automatically applied by DataType in separate files. You need to define them all in app.xaml or the corresponding window, which is not very useful. that would also remove the need to define a ViewLocator as from the default samples as you can stick with automatically loaded datatemplates by datatype.

**Comment by yll690** [comment]
I have a new idea. How about adding a `DataTemplates` property to `Styles`? Avalonia can also search every `Styles.DataTemplates` like searching `Application.DataTemplates` and `Controls.DataTemplates`. Then we can use `StyleInclude` instead of `DataTemplateInclude` and the `DataTemplate` can apply to models automatically, too.
